### PR TITLE
New analysis: standard errors of measurement 

### PR DESCRIPTION
**Pull request by juliuspfadt**
this adds standard error of measurement estimates. 
I need the PR for the submission of a tutorial paper. 

I dont really know why the snapshots in the tests for the other analysis are added.
The test will likely fail because of the dynamic qml elements. I hope @vandenman has a remedy to this with jaspTools

### CODE REVIEW

File: tests/testthat/test-standardErrorOfMeasurement.R [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  )
+)
+
+results <- runAnalysis("standardErrorOfMeasurement", data = "asrm.csv", options = options)
+# results <- runAnalysis("standardErrorOfMeasurement", dataset = Bayesrel::asrm, options = options, makeTests = TRUE)
+
+
+
+# binary data
+options <- analysisOptions("standardErrorOfMeasurement")
```
Comment (vandenman) [inline tests/testthat/test-standardErrorOfMeasurement.R:45]: please change this to directly specify the options list, jaspTools cannot handle this.

File: R/standardErrorOfMeasurement.R [hunk 2]
```diff
@@ -0,0 +1,890 @@
...
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+#' @export
+standardErrorOfMeasurement <- function(jaspResults, dataset, options) {
+
+
+# sink(file = "~/Downloads/log.txt")
+# on.exit(sink(NULL))
+
```
Comment (vandenman) [inline R/standardErrorOfMeasurement.R:24]:
```suggestion
```

File: tests/testthat/test-standardErrorOfMeasurement.R [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  plotHeight = 320,
+  plotWidth = 480,
+  pointPlots = TRUE,
+  reliabilityValue = 0.5,
+  thorndike = TRUE,
+  userReliability = FALSE,
+  variables = list(
+    types = c("ordinal", "ordinal", "ordinal", "ordinal", "ordinal"),
+    value = c("asrm_1", "asrm_2", "asrm_3", "asrm_4", "asrm_5")
+  )
```
Comment (vandenman) [inline tests/testthat/test-standardErrorOfMeasurement.R:36]: should be
```suggestion
  variables = c("asrm_1", "asrm_2", "asrm_3", "asrm_4", "asrm_5"),
  variables.types = c("ordinal", "ordinal", "ordinal", "ordinal", "ordinal")
```

**Comment by juliuspfadt** [comment]
I changed the tests a bit, however, I cannot make tests from the test-file. It always gives an error when `makeTests=TRUE`

**Review by vandenman — CHANGES_REQUESTED** [review]
- The analysis does not like `contExpon`:

image

or continuous variables in the debug dataset in general:

image

image

- Is there a better name than "combined plot"? 

Using the dataset below I noticed that:

- Mollenkopf-Feldt is empty for some reason:

image

perhaps add a footnote?

- The 2PL plot looks incorrect (but that could just be me)

image


[reliability_pr_210_debug.jasp.zip](https://github.com/user-attachments/files/16749831/reliability_pr_210_debug.jasp.zip)

**Comment by juliuspfadt** [comment]
hmm, well continuous variables are not allowed. It's easy to force an ordinal variable to be continuous but not the other way around. We have allowed columns, but that is by now only a suggestion. I dont know how to remedy this. If allowed columns would be strict...

**Comment by juliuspfadt** [comment]
Have you found a way to create the unit tests?

**Comment by juliuspfadt** [comment]
Alright, thanks. I have changed the levels and the tests. Can you review again?

**Comment by juliuspfadt** [comment]
I checked the pot again, and now it looks like this: 
Screenshot 2024-09-23 at 15 33 08
I checked the numbers in R and I think the issue is that with that data the IRT model yields some weird results; it also takes more iterations than "usual". I am inclined to leave it as is

**Review by vandenman — APPROVED** [review]

**Comment by vandenman** [comment]
Looks good! Do you think this is a good point to maybe start a `NEWS.md` file for this module? See e.g., [here](https://github.com/r-lib/ragg/blob/main/NEWS.md) for an example, this does not need to be long and the goal is to facilitate the automatic generation of release notes. We want to do this for a longer time, but I'd first like to try it out with one or two modules to see how well it goes (see also https://github.com/jasp-stats/INTERNAL-jasp/issues/1913).

**Pull request merged**